src/main/java/com/tabnine/inline/InlineCompletionCache.kt

```diff
@@ -0,0 +1,39 @@
+package com.tabnine.inline
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.Key
+import com.tabnine.prediction.TabNineCompletion
+import java.util.stream.Collectors
+
+class InlineCompletionCache {
+
+    companion object {
+        @JvmStatic
+        val instance = InlineCompletionCache()
+
+        private val INLINE_COMPLETIONS_LAST_RESULT = Key.create<List<TabNineCompletion>>("INLINE_COMPLETIONS_LAST_RESULT")
+    }
+
+    fun store(editor: Editor, completions: List<TabNineCompletion>) {
+        editor.putUserData(INLINE_COMPLETIONS_LAST_RESULT, completions)
+    }
+
+    fun retrieveAdjustedCompletions(editor: Editor, userInput: String): List<TabNineCompletion> {
```

Comment:
maybe add an early return if the editor == null? i know this shouldn't happen in theory, but I won't be surprised if there's an edge case where this is null, and we'll bombard a poor user with exceptions